my_enumerable_methods.rb

```diff
@@ -0,0 +1,138 @@
+module Enumerable
+  def my_each
+    if block_given?
+      each do |counter|
+        yield(counter)
+      end
+    else
+      to_enum(:my_each)
+    end
+  end
+
+  def my_each_with_index
+    if block_given?
+      (0...size).each do |counter|
+        yield self[counter], counter
+      end
+    else
+      to_enum(:my_each_with_index)
+    end
+  end
+
+  def my_select
+    select_in = []
+    my_each do |counter|
+      select_in << yield(counter)
+    end
+    select_in
+  end
+
+  def my_all?
+    my_each do |element|
+      if block_given?
+        yield(element)
+      elsif element.nil? or !i or !yield(element)
+        puts false
+      end
+    end
+    puts true
+  end
+
+  def my_any?
+    my_each do |element|
+      if block_given?
+        yield element
+      elsif element.nil? or element or yield(element)
+        puts true
+      end
+    end
+    puts false
+  end
```

Comment:
- [ ] should be identical and returns the same thing as ruby's [`any?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-any-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_any.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- [ ] when no block or argument is given returns true if at least one of the collection is not false or nil
```ruby
true_array = [nil, false, true, []]
true_array.my_any? == true_array.any? #true
```
- [ ]  when a class is passed as an argument returns true if at least one of the collection is a member of such class
```ruby
words = %w[dog door rod blade]
words.my_any?(Integer) == words.any?(Integer) #true
```
- [ ] when a Regex is passed as an argument returns false if none of the collection matches the Regex
```ruby
words.my_any?(/z/) == words.any?(/z/) #true
```
- [ ] when a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern
```ruby
words.my_any?('cat') == words.any?('cat') #true
```
- [ ] returns true if the block ever returns a value other than false or nil
```ruby
true_block = proc { |num| num <= 9 } 
false_block = proc { |num| num > 9 } 
array.my_any?(&true_block) == array.any?(&true_block) #true
array.my_any?(&false_block) == array.any?(&false_block) #true
```